parlai/core/torch_classifier_agent.py

```diff
@@ -147,7 +147,8 @@ def __init__(self, opt, shared=None):
                     raise ValueError(
                         'Cannot combine --data-parallel and distributed mode'
                     )
-                self.model = torch.nn.DataParallel(self.model)
+                if shared is None:
```

Comment:
let's get rid of this `if` here and put this entire `if self.use_cuda` block under the `else:` statement above

Reply:
Let's actually do this one in this PR.